src/main/java/com/exasol/errorreporting/PlaceholderMatcher.java

```diff
@@ -0,0 +1,72 @@
+package com.exasol.errorreporting;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An iterable for placeholder in a given text.
+ *
+ */
+public class PlaceholderMatcher implements Iterable<Placeholder> {
+    private final String text;
+
+    private PlaceholderMatcher(final String text) {
+        this.text = text;
+    }
+
+    /**
+     * Find placeholders in a given string.
+     * 
+     * @param text string containing placeholders
+     * @return iterable for placeholders
+     */
+    public static PlaceholderMatcher findPlaceholders(final String text) {
+        return new PlaceholderMatcher(text);
+    }
+
+    @Override
+    public Iterator<Placeholder> iterator() {
+        return new PlaceholderIterator(this.text);
+    }
+
+    private static class PlaceholderIterator implements Iterator<Placeholder> {
+        private static final Pattern PLACEHOLDER_PATTERN = Pattern.compile("\\{\\{([^\\}]*)\\}\\}");
```

Comment:
I think now we can move the `PLACEHOLDER_PATTERN` to the `PlaceholderMatcher`, and have it as parameter in the `PlaceholderMatcher` constructor, as in:

```
 private PlaceholderIterator(final String text, final Pattern pattern) {
            this.matcher = pattern.matcher(text);
            this.hasNext = this.matcher.find();
        }
```

Reply:
parameters that are not variable cause warnings in my IDE. So not sure if it's a good idea